### DISCUSSION TITLE
When importing accounts using private key, phantom is throwing -32603 when trying to connect not imported ones.

### DISCUSSION THREAD

**Discussion by mojtabast** [phantom/docs#302] [Help & Support]
### Summary

If user enabled multi chains for its wallet (Solana, Ethereum, Polygon), Phantom is injecting `solana` and `ethereum` into `window`. But if a user added an account which is only one of these (for example, only Ethereum) using `Import Private Key` option, As an dApp developer, we can not detect available networks since both of `.solana` and `.ethereum` are injected.

In this scenario, I think phantom should do one of these:
1. Its behavior should be like when a network is disabled in setting and `Active Networks` which means if a network is not available, instance shouldn't be injected into `window`.
2. If it's desired behavior, Phantom should throw an specific error, other than `-32603`  to let developers detect this case and handle it accordingly.

   

### Steps to Reproduce

1. Enable all the networks (Solana, EVM).
2. Try to add a new account using `Import Private Key` option.
3. Import a Private Key for **only** Ethereum.
4. try to run `window.phantom.solana.connect()`. it throw an -32603.

### Is there an existing discussion for this?

- [X] I have searched the existing discussions

**Comment by adamdelphantom** [comment]
Hey @mojtabast,

Thanks for writing in!

We are aware of a bug where the ethereum provider is injected when it shouldn't be. I don't have an eta on the fix.
